controller/k8s/watcher.go

```diff
@@ -36,6 +47,19 @@ func (w *watcher) run() error {
 	initialized := make(chan struct{}, 1)
 	defer close(initialized)
 
+	go wait.ExponentialBackoff(wait.Backoff{
+		Duration: durationBetweenBackoff,
+		Factor:   watchBackoffFactor,
+		Steps:    backoffSteps,
+	}, func() (bool, error) {
+		err := w.watchInit(w.stopCh)
+		if err != nil {
+			log.Errorf("Error establishing watch in [%s watcher]. Retrying", w.resourceType)
+			return false, nil
+		}
+		return true, err
```

Comment:
Based on my reading of the [ExponentialBackoff docs](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/wait/wait.go#L182), returning true here will cause us to stop retrying, but we actually want to retry indefinitely, or at least until MaxInt. So I'd change this to:

```
		return false, err
```